packages/react-devtools-shared/src/backend/agent.js

```diff
@@ -310,20 +313,16 @@ export default class Agent extends EventEmitter<{|
   }
 
   getIDForNode(node: Object): number | null {
+    const renderers = [];
     for (const rendererID in this._rendererInterfaces) {
       const renderer = ((this._rendererInterfaces[
         (rendererID: any)
       ]: any): RendererInterface);
-
-      try {
-        const id = renderer.getFiberIDForNative(node, true);
-        if (id !== null) {
-          return id;
-        }
-      } catch (error) {
-        // Some old React versions might throw if they can't find a match.
-        // If so we should ignore it...
-      }
+      renderers.push(renderer);
+    }
+    const rendererInterface = getBestMatchingRendererInterface(renderers, node);
```

Comment:
This probably should be in the `backend/agent` file instead of the utils because you're using two for loops right now and if you put the code from `getBestMatchingRendererInterface` into the agent you would only need one

ex. 

```js
let renderer = this._rendererInterfaces[0];
for (const rendererID in this._rendererInterfaces) {
  const currentRenderer = ...;
  const fiber = currentRenderer.getFiberForNative(node, false);
  if (fiber !== null && fiber.stateNode === node) {
   renderer = currentRenderer;
    break;
  }
  ...
}
...
```

Reply:
The first for-loop is only there for proper typing. Maybe I should refactor the types of `this._rendererInterfaces`?
It's indeed easier to do everything `agent`, but how about the use of the util function in `Overlay`?

Reply:
I updated the file with better typing https://github.com/facebook/react/pull/24665/commits/d0346ff11517296b2e0f95d8a64e1802d3fbfc13 so that I can use `values()` instead of for-loop https://github.com/facebook/react/pull/24665/commits/9a7e0beb41f6cec0a85634a23bd92efc2084293b

Does it make sense?